nrf_security/cmake/kconfig_mbedtls_configure.cmake

```diff
@@ -7,55 +7,207 @@
 # on the Kconfig settings.
 # When all values has been processed, the mbedtls config file will be generated.
 
+
+#
+# Internal macro that shows debug output in for
+# nrf_security related configurations when
+# Cmake is run with -DDEBUG_KCONFIG_MBEDTLS=1
+#
+macro(kconfig_mbedtls_debug)
+  if(DEBUG_KCONFIG_MBEDTLS)
+    message(STATUS ${ARGN})
+  endif()
+endmacro()
+
+
+#
 # Internal macro which will enable the define in mbed TLS config file based on
 # the correspong Kconfig setting.
+# It will additionally set CONFIG_VANILLA_ONLY_MBEDTLS_XXXX in case no other
+# backends is enabling the same feature
+#
+macro(kconfig_mbedtls_config base)
+  if (CONFIG_${base} OR
+      CONFIG_CC310_${base} OR
+      CONFIG_VANILLA_${base})
+      kconfig_mbedtls_debug("Setting ${base} to TRUE")
+    set(${base} TRUE)
+  endif()
+  if ((CONFIG_${base} AND NOT CONFIG_CC310_BACKEND) OR
+      (CONFIG_VANILLA_${base} AND NOT CONFIG_CC310_${base}))
+      kconfig_mbedtls_debug("Setting CONFIG_VANILLA_ONLY_${base} to TRUE")
+    set(CONFIG_VANILLA_ONLY_${base} TRUE)
+  endif()
+  if (CONFIG_${base} AND NOT CONFIG_MBEDTLS_VANILLA_BACKEND)
+    kconfig_mbedtls_debug("Setting CONFIG_CC310_${base}nin to TRUE")
+    set(CONFIG_CC310_${base} TRUE)
```

Comment:
The CONFIG_ namespace is reserved.

It is not permitted to set variables in this namespace.

Reply:
Fixed